homeassistant/components/upnp/__init__.py

```diff
@@ -150,8 +157,22 @@ def device_discovered(info: Mapping[str, Any]) -> None:
         model=device.model_name,
     )
 
+    update_interval_sec = entry.options.get(
+        CONFIG_ENTRY_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+    )
+    update_interval = timedelta(seconds=update_interval_sec)
+    LOGGER.debug("update_interval: %s", update_interval)
+    coordinator = UpnpDataUpdateCoordinator(
```

Comment:
You should store the coordinator instead of the device in hass.data

Reply:
To clarify on this, remove:
```
    # Save device.
    hass.data[DOMAIN][DOMAIN_DEVICES][udn] = device
```
correct? I added the following to store the coordinator in hass.data:
```
    # Save coordinator.
    hass.data[DOMAIN][entry.entry_id] = coordinator
```

But did not remove storing device as I was not sure if that is also used somewhere else within HASS for devices. Just want to clarify before I remove. :-)

Reply:
That is what I meant :+1: `hass.data[DOMAIN][DOMAIN_DEVICES][udn]` was only used by the `sensor.py` file to get the device, but we do not get that anymore, we get the coordinator instead :+1: